pkg/codegen/dotnet/utilities.go

```diff
@@ -78,3 +80,45 @@ func makeValidIdentifier(name string) string {
 func propertyName(name string) string {
 	return makeValidIdentifier(Title(name))
 }
+
+func makeSafeEnumName(name string) (string, error) {
+	safeName := name
+
+	// If the name is one illegal character, replace it.
+	if len(safeName) == 1 && !isLegalIdentifierStart(rune(safeName[0])) {
+		enumReplacer := strings.NewReplacer(
+			"0", "Zero",
+			"1", "One",
+			"2", "Two",
+			"3", "Three",
+			"4", "Four",
+			"5", "Five",
+			"6", "Six",
+			"7", "Seven",
+			"8", "Eight",
+			"9", "Nine",
+			"*", "Asterisk",
+		)
+
+		safeName = enumReplacer.Replace(safeName)
+
+		// If it's still an illegal character (we weren't able to find a replacement), return an error.
+		if !isLegalIdentifierStart(rune(safeName[0])) {
+			return "", fmt.Errorf("enum name %s is not a valid identifier", safeName)
+		}
+	}
+
+	// Capitalize and make a valid identifier.
+	safeName = strings.Title(makeValidIdentifier(safeName))
+
+	// If there are multiple underscores in a row, replace with one.
+	regex := regexp.MustCompile(`_+`)
+	safeName = regex.ReplaceAllString(safeName, "_")
+
+	// "Equals" conflicts with a method on the EnumType struct, change it to EqualsValue.
+	if safeName == "Equals" {
+		safeName = fmt.Sprintf("%sValue", safeName)
```

Comment:
Nit: Any reason not to do this?

```suggestion
		safeName = "EqualsValue"
```

Reply:
I left it flexible in case there are other values like "Equals" that would cause issues but no reason not to simplify since there aren't at the moment.